test/scheduler_test.rb

```diff
@@ -456,6 +456,52 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      def self.before_enqueue_example(*)
+        if Resque.redis.setnx('before_enqueue_job:enqueued', 'true')
+          sleep 5 # sleep longer than test timeout
```

Comment:
Hmm I think I would use a class variable here to know when to continue instead of relying on the timeout. Below, you could do something like:
```ruby
pid = fork do
  trap("TERM") { BeforeEnqueueJob.received_shutdown_signal = true }
  Resque.schedule = schedule
  Resque::Scheduler.run
end
```
and then here do `sleep 0.1 until received_shutdown_signal`.

Not sure if that's much better but it seems less flaky and less time-dependent.